.github/workflows/check-cla.yaml

```
@@ -0,0 +1,16 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: cla-check
+on:
+  workflow_call:
+  pull_request:
+
+jobs:
+  cla-check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check if Canonical CLA has been signed
+        uses: canonical/has-signed-canonical-cla@v1
+        with:
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
```

[nit]: Missing newline at the end of file:

```suggestion
          github-token: "${{ secrets.GITHUB_TOKEN }}"
          
```